utils/logger/logger.go

```diff
@@ -0,0 +1,205 @@
+// this is a new logger interface for mattermost
+
+package logger
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"runtime"
+
+	l4g "github.com/alecthomas/log4go"
+
+	"github.com/mattermost/platform/model"
+	"github.com/mattermost/platform/utils"
+)
+
+// this pattern allows us to "mock" the underlying l4g code when unit testing
+var debug = l4g.Debug
+var info = l4g.Info
+var err = l4g.Error
+
+func init() {
+	initL4g(utils.Cfg)
+}
+
+// listens for configuration changes that we might need to respond to
+var configListenerID = utils.AddConfigListener(func(oldConfig *model.Config, newConfig *model.Config) {
+	info("Configuration change detected, reloading log settings")
+	initL4g(newConfig)
+})
+
+// assumes that ../config.go::configureLog has already been called, and has in turn called l4g.close() to clean up
+// any old filters that we might have previously created
```

Comment:
Eventually, the log initialization code in `../config.go` will probably be moved entirely over to here, but I need to leave it in place until this code entirely replaces all existing l4g code.